### DISCUSSION TITLE
Ztunnel and RDS Aurora Postgres - Connection reset error logs

### DISCUSSION THREAD

**Discussion by jbye** [istio/istio#52600] [Q&A] [answered]
We are on Istio 1.22.3 with ambient mode enabled.

We are seeing a relative large amount of error level log messages of the following pattern in our ztunnel logs:

```
2024-08-09T12:33:17.918809Z    error    access    connection complete    src.addr=xxx:55216 src.workload="xxx-xxx-xxx" src.namespace="xxx" dst.addr=xxx:5432 direction="outbound" bytes_sent=1343 bytes_recv=3858 duration="58965ms" error="io error: Connection reset by peer (os error 104)"
```

These all relate to connections to RDS Aurora Postgres. With reference to the discussion in this PR, https://github.com/kubernetes/kubectl/issues/1169, this seems to be expected behavior of RDS Aurora.

> I found that the reason is that the Postgres server sends an RST packet when the client(like psql) disconnecting from the server in TLS mode, because it shut down the sub-process that handling connection without reading the SSL Shutdown packet send from the client.

Since this appears to be expected behavior by RDS Aurora when SSL is enabled, how can we prevent the Ztunnel error logs from being spammed with this noise? I assume there is no built-in way to not log these specific events.

**Comment by howardjohn** [comment, marked as answer]
This will be fixed in 1.22.4. The fix is simply hiding the logs, so while annoying it shouldn't impact the traffic. https://github.com/istio/ztunnel/pull/1219

**Reply by jbye** [reply]
That's great. Thanks for the quick response!
